assets/js/components/GoogleChart/utils.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * Google Chart utilities.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import set from 'lodash/set';
+import cloneDeep from 'lodash/cloneDeep';
+import merge from 'lodash/merge';
+
+/**
+ * Internal dependencies
+ */
+import { stringToDate } from '../../util/date-range/string-to-date';
+
+/**
+ * Returns the Google chart data, filtered by selected stats if present.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array|undefined} data          Chart data.
+ * @param {Array|undefined} selectedStats The columns that should be displayed for the data set.
+ * @return {Object} The chart data, filtered by selected stats if present.
+ */
+export const getFilteredChartData = ( data, selectedStats ) => {
+	if ( ! selectedStats?.length ) {
+		return data;
+	}
+	// Ensure we don't filter out columns that aren't data, but are things like
+	// tooltips or other content.
+	let nonDataColumns = [];
+	if ( data?.length ) {
+		nonDataColumns = data[ 0 ].reduce( ( acc, row, rowIndex ) => {
+			return row?.role ? [ ...acc, rowIndex ] : acc;
+		}, [] );
+	}
+
+	let filteredData = data;
+	if ( selectedStats?.length > 0 ) {
+		filteredData = data.map( ( row ) => {
```

Comment:
As we're bailing early when `selectedStats?.length` is falsy, we don't need to check it again here:

```suggestion
	return data.map( ( row ) => {
```